SecretManagement.LastPass.Extension/SecretManagement.LastPass.Extension.psm1

```diff
@@ -61,20 +88,55 @@ function Get-Secret
         $Name = $Matches[1]
     }
 
-    $res = Invoke-lpass 'show','--name', $Name, '--password'
-    if ([string]::IsNullOrWhiteSpace($res)) {
-        $res = Invoke-lpass 'show', '--name', $Name, '--notes'
-    } else {
-        # We have a password, check for a username
-        $username = Invoke-lpass 'show', '--name', $Name, '--username'
-        if ($username) {
-            $res = [System.Management.Automation.PSCredential]::new(
-                $username,
-                (ConvertTo-SecureString $res -AsPlainText -Force))
+    try {
+        $res = Invoke-lpass 'show', '--name', $Name, '--all'
+        if ($null -eq $res) {
+            return 
+        }
+        if ($res[0] -eq 'Multiple matches found.') {
+            Write-Warning "Multiple matches found with the name $Name. `nThe first matching result will be returned."
+            $Id = [regex]::Match($res[1], '\[id: (.*)\]').Groups[1].value
+            $res = Invoke-lpass 'show', '--name', $Id, '--all'
+
         }
     }
+    catch {
+        Write-Error $_
+        return 
+    }
+    
+    $Raw = ($res | Select-Object -Skip 1) -join "`n"
 
-    return $res
+    if ([String]::IsNullOrEmpty($Raw)) {
+        return ""
+    }
+
+    # Custom type cannot have the same case-sensitive name but Username / uSERname work.
+    # That's why we don't parse it directly to a hashtable.
+    $MyMatches = @([regex]::Matches($raw, '(?<key>.*?)\: (?<value>.*?)(?:\n|$)')  | ForEach-Object {
+            [PSCustomObject]@{
+                key        = $_.Groups.Item('key').value
+                value      = $_.Groups.Item('value').value 
+                valueIndex = $_.Groups.Item('value').index
+            }
+        })
+
+    # Notes is always the last item. This is also the only field that can be multiline.
+    $HasNote = $MyMatches.key -ccontains 'Notes' 
```

Comment:
@TylerLeonhardt  Consider the following. 

- Notes field can exist only once.
- Notes field can be multiline
- Notes field can contains something like : 

Notes:
```
Here are the notes.
Nothing so tricky
Notes: This is tricky
Pin: 1234
```

The `$MyMatches` regex pattern `(?<key>.*?)\: (?<value>.*?)(?:\n|$)` will pick up the legitimate fields but also the  second
`Notes: This is tricky` and `Pin: 1234` 

Therefore, while `Notes:` is always the last field, `$MyMatches` will contains records that match the pattern after the real notes, which could include fake `Notes` key / value pair. 

Thus, the first `Notes` item found is to be considered to avoid issues with that potential edge case. 
The first `Notes` will always be the last REAL match. 

This is why I discard any matches after Notes, since they are just things that match the pattern but actually belong to the same last multiline Notes item.
```
 $MyMatches = $MyMatches.Where({$_.ValueIndex -lt $start})
```

I added the following comment in the code to make this clearer.

```
# Notes is always the last item (lpass wise). This is also the only field that can be multiline.
# Any matching items after the first "Notes" key need to be discarded as it is not an item,
# but rather just part of the notes.
```

Reply:
That makes perfect sense! Thanks!